src/main.rs

```diff
@@ -73,9 +78,30 @@ fn main() {
 
     let host = args.flag_host.unwrap_or("0.0.0.0".to_string());
     let port = args.flag_port.unwrap_or(4242);
+    let using_tls = args.flag_cert_directory.is_some();
+
+    let iron = Iron::new(chain);
     info!("Starting server on {}:{}", host, port);
-    Iron::new(chain).http(format!("{}:{}", host, port).as_ref() as &str)
-        .unwrap();
+    let addr = format!("{}:{}", host, port);
+
+    if !using_tls {
+        iron.http(addr.as_ref() as &str)
+            .unwrap();
+    } else {
+        info!("Starting TLS server");
+        let certificate_directory = args.flag_cert_directory.unwrap();
+        let certificate_directory = PathBuf::from(certificate_directory);
+
+        let mut private_key = certificate_directory.clone();
+        private_key.push("privkey.pem");
+
+        let mut cert = certificate_directory.clone();
+        cert.push("cert.pem");
```

Comment:
what about the chain cert? I'm seeing an error when trying `openssl s_client -connect knilxof.org:4443` (which runs this code) but not when trying `openssl s_client -connect knilxof.org:4243` (which runs node proxy.js). Also, that reminds me, we can remove proxy.js from this repo once this is resolved.